src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -133,6 +129,14 @@ private static List<String> finalizeParameterNames(List<String> parameterNames,
             result.add("x" + i);
         }
 
+        if (needsAllocator) {
+            String allocatorName = "allocator";
+            while (result.contains(allocatorName)) {
+                allocatorName = "_" + allocatorName;
```

Comment:
In other places we've used `$` suffixes to deconflict names. Should we do the same here?

Reply:
I did this because the name mangler uses `_` when a name is a keyword e.g. `var`.